rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -64,17 +64,15 @@ namespace rsx
 			rsx->sync_point_request = true;
 			const u32 addr = get_address(method_registers.semaphore_offset_406e(), method_registers.semaphore_context_dma_406e());
 
-#ifdef IS_LE_MACHINE
-			arg = se_storage<u32>::swap(arg);
-			const auto& sema = vm::_ref<le_t<u32>>(addr);
-#else
+			// Preswap value
+			arg = static_cast<u32>(be_t<u32>(arg));
```

Comment:
This expression doesn't actually swap bytes. I recommended to create a new (const) variable.

Reply:
Neither methods was working as expected, so I used raw() to make a one-direction swap.
I'm open to make this a method from se_storage like se_storage<>::get<bool is_be>.